azurerm/resource_arm_virtual_network_gateway.go

```diff
@@ -97,11 +97,24 @@ func resourceArmVirtualNetworkGateway() *schema.Resource {
 				// and validateArmVirtualNetworkGatewayExpressRouteSku.
 				ValidateFunc: validation.Any(
 					validateArmVirtualNetworkGatewayPolicyBasedVpnSku(),
-					validateArmVirtualNetworkGatewayRouteBasedVpnSku(),
+					validateArmVirtualNetworkGatewayRouteBasedVpnSkuGeneration1(),
+					validateArmVirtualNetworkGatewayRouteBasedVpnSkuGeneration2(),
 					validateArmVirtualNetworkGatewayExpressRouteSku(),
 				),
 			},
 
+			"generation": {
+				Type:     schema.TypeString,
+				Default:  string(network.VpnGatewayGenerationGeneration1),
```

Comment:
I think we need to remove this default - since it causes at least one of the tests to fail:

>         Error: Error Creating/Updating AzureRM Virtual Network Gateway "acctestvng-191217203404893534" (Resource Group "acctestRG-191217203404893534"): network.VirtualNetworkGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="VirtualNetworkGatewayTypeDoesNotSupportVpnGatewayGeneration" Message="The VirtualNetworkGatewayType ExpressRoute does not support VirtualNetworkVpnGatewayGeneration Generation1. Only VirtualNetworkVpnGatewayGeneration None is supported by VirtualNetworkGatewayType ExpressRoute." Details=[]

Reply:
I hope you don't mind, but so that we can get this merged I'm going to push a commit which fixes this 👍